src/MUIDataTable.js

```diff
@@ -196,8 +196,16 @@ class MUIDataTable extends React.Component {
   }
 
   componentDidUpdate(prevProps) {
-    if (this.props.data !== prevProps.data || this.props.columns !== prevProps.columns) {
-      this.setTableData(this.props, TABLE_LOAD.INITIAL, () => {
+    if (
+      this.props.data !== prevProps.data ||
+      this.props.columns !== prevProps.columns ||
+      this.props.options !== prevProps.options
+    ) {
+      if (this.props.options !== prevProps.options) {
+        this.getDefaultOptions(this.props);
+        this.setTableOptions(this.props);
+      }
+      this.setTableData(prevProps, TABLE_LOAD.INITIAL, () => {
```

Comment:
Using `prevProps` here opens this regression: https://github.com/gregnb/mui-datatables/pull/597.